framework/Functions.py

```diff
@@ -61,59 +82,50 @@ def __init__(self,runInfoDict):
-                      '__residuum__','__residuum','residuum','__gradient__','__gradient','gradient']:
-          if method in ['__residuumSign__','__residuumSign','residuumSign']:
-            self.__residuumSign                                =  importedModule.__dict__[method]
-            self.__actionDictionary['residuumSign' ]           = self.__residuumSign
-            self.__actionImplemented['residuumSign']           = True
-          if method in ['__supportBoundingTest__','__supportBoundingTest','supportBoundingTest']:
-            self.__supportBoundingTest                         =  importedModule.__dict__[method]
-            self.__actionDictionary['supportBoundingTest' ]    = self.__supportBoundingTest
-            self.__actionImplemented['supportBoundingTest']    = True
-          if method in ['__residuum__','__residuum','residuum']:
-            self.__residuum                                    =  importedModule.__dict__[method]
-            self.__actionDictionary['residuum' ]               = self.__residuum
-            self.__actionImplemented['residuum']               = True
-          if method in ['__gradient__','__gradient','gradient']:
-            self.__gradient                                    =  importedModule.__dict__[method]
-            self.__actionDictionary['gradient']                = self.__gradient
-            self.__actionImplemented['gradient']               = True
-        else:
-          #custom
-          self.__actionDictionary[method]                    = importedModule.__dict__[method]
-          self.__actionImplemented[method]                   = True
-    else:
-      self.raiseAnError(IOError,'No file name for the external function has been provided for external function '+self.name+' of type '+self.type)
-    cnt = 0
-    for child in xmlNode:
-      if child.tag=='variable':
-        execCommand('self.'+child.text+' = None',self=self)
-        self.__inputVariables.append(child.text)
-        cnt +=1
-        if len(child.attrib.keys()) > 0:
-          self.raiseAnError(IOError,'variable block in the definition of the function '+self.name + ' should not have any attribute!')
-    if cnt == 0:
-      self.raiseAnError(IOError,'not variable found in the definition of the function '+self.name)
+    self.functionFile = paramInput.parameterValues["file"]
+    # get the module to load and the filename without path
+    moduleToLoadString, self.functionFile = utils.identifyIfExternalModelExists(self, self.functionFile, self.workingDir)
+    # import the external function
+    importedModule = utils.importFromPath(moduleToLoadString,self.messageHandler.getDesiredVerbosity(self)>1)
+    if not importedModule:
+      self.raiseAnError(IOError,'Failed to import the module '+moduleToLoadString+' supposed to contain the function: '+self.name)
+    #here the methods in the imported file are brought inside the class
+    for method in importedModule.__dict__.keys():
```

Comment:
good advice...fixed and removed the link to self.__residuum etc because not used (actually I am planning in another merge request to remove that dictionary directly bacause not needed)